### PR TITLE
Roll bundle with prebuilt

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: packages/firestore/rollup.config.browser.js [hunk 1]
```diff
@@ -34,12 +38,49 @@ export default [
...
+    plugins: [
+      alias({
+        entries: [
+          {
+            find: /^(.*)\/export$/,
+            replacement: `$1\/dist/prebuild.js`
+          }
+        ]
+      }),
+      ...util.es2017Plugins('browser', /* mangled= */ false)
```
Comment (schmidt-sebastian) [inline packages/firestore/rollup.config.browser.js:60]: This should be mangled.

File: packages/firestore/rollup.config.browser.js [hunk 2]
```diff
@@ -34,12 +38,49 @@ export default [
...
+    },
+    output: [
+      {
+        dir: 'dist/esm5',
+        format: 'es',
+        sourcemap: true
+      }
+    ],
+    plugins: util.es2017ToEs5Plugins(),
+    external: util.resolveNodeExterns,
```
Comment (schmidt-sebastian) [inline packages/firestore/rollup.config.browser.js:81]: This should be `resolveBrowserExterns`.

File: packages/firestore/rollup.config.rn.js [hunk 3]
```diff
@@ -15,20 +15,48 @@
...
+    plugins: [
+      alias({
+        entries: [
+          {
+            find: /^(.*)\/export$/,
+            replacement: `$1\/dist/prebuild.rn.js`
+          }
+        ]
+      }),
+      ...util.es2017Plugins('rn', /* mangled= */ false)
```
Comment (schmidt-sebastian) [inline packages/firestore/rollup.config.rn.js:55]: This should be mangled.

File: packages/firestore/rollup.config.browser.js (hunk 1 above)
Comment (wu-hui) [reply packages/firestore/rollup.config.browser.js:54]: As discussed offline, this should be false.

File: packages/firestore/rollup.config.rn.js (hunk 3 above)
Comment (wu-hui) [reply packages/firestore/rollup.config.rn.js:48]: Same as browser build, this should be false.

File: packages/firestore/rollup.config.rn.js [hunk 4]
```diff
@@ -15,20 +15,61 @@
...
+      bundle: 'index.bundle.ts'
+    },
+    output: {
+      dir: 'dist/rn',
+      format: 'es',
+      sourcemap: true
+    },
+    plugins: [
+      util.applyPrebuilt('prebuilt.rn.js'),
+      ...util.es2017Plugins('rn', /* mangled= */ true)
```
Comment (schmidt-sebastian) [inline packages/firestore/rollup.config.rn.js:68]: This should also not be mangled.

File: packages/firestore/src/core/bundle_types.ts [hunk 5]
```diff
@@ -26,15 +25,8 @@ import { SnapshotVersion } from './snapshot_version';
  * Represents a Firestore bundle saved by the SDK in its local storage.
  */
 export interface Bundle {
-  /**
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/bundle_types.ts:29]: Please add back.

File: packages/firestore/src/core/bundle_types.ts [hunk 6]
```diff
@@ -26,15 +25,8 @@ import { SnapshotVersion } from './snapshot_version';
...
-   * Id of the bundle. It is used together with `createTime` to determine if a
-   * bundle has been loaded by the SDK.
-   */
   readonly id: string;
-
-  /** Schema version of the bundle. */
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/bundle_types.ts:35]: Please add back.

File: packages/firestore/src/core/bundle_types.ts [hunk 7]
```diff
@@ -26,15 +25,8 @@ import { SnapshotVersion } from './snapshot_version';
...
   readonly version: number;
-
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/bundle_types.ts:37]: Please add back.

File: packages/firestore/src/core/bundle_types.ts [hunk 8]
```diff
@@ -46,9 +38,7 @@ export interface Bundle {
  * Represents a Query saved by the SDK in its local storage.
  */
 export interface NamedQuery {
-  /** The name of the query. */
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/bundle_types.ts:49]: Please add back.

File: packages/firestore/src/core/bundle_types.ts [hunk 9]
```diff
@@ -46,9 +38,7 @@ export interface Bundle {
...
   readonly name: string;
-  /** The underlying query associated with `name`. */
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/bundle_types.ts:51]: Please add back.

**Review by schmidt-sebastian — COMMENTED** [review]
Last round, but basically LGTM

File: packages/firestore/src/core/bundle_types.ts (hunk 5 above)
Comment (wu-hui) [reply packages/firestore/src/core/bundle_types.ts]: Done.

File: packages/firestore/src/core/bundle_types.ts (hunk 6 above)
Comment (wu-hui) [reply packages/firestore/src/core/bundle_types.ts]: Done.

File: packages/firestore/src/core/bundle_types.ts (hunk 7 above)
Comment (wu-hui) [reply packages/firestore/src/core/bundle_types.ts]: Done.

File: packages/firestore/src/core/bundle_types.ts (hunk 8 above)
Comment (wu-hui) [reply packages/firestore/src/core/bundle_types.ts]: Done.

File: packages/firestore/src/core/bundle_types.ts (hunk 9 above)
Comment (wu-hui) [reply packages/firestore/src/core/bundle_types.ts]: Done.

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**